engine/class_modules/sc_death_knight.cpp

```diff
@@ -8601,14 +8677,26 @@ void death_knight_t::create_buffs()
       -> set_affects_regen( true )
       -> set_stack_change_callback( [ this ]( buff_t*, int, int )
            { _runes.update_coefficient(); } );
+
   buffs.frenzied_monstrosity = make_buff( this, "frenzied_monstrosity", find_spell ( 334896 ) )
     -> add_invalidate( CACHE_ATTACK_SPEED )
     -> add_invalidate( CACHE_PLAYER_DAMAGE_MULTIPLIER );
 
+  buffs.final_sentence = make_buff( this, "final_sentence", find_spell( 353823 ) )
+    -> set_default_value_from_effect_type( A_MOD_DAMAGE_PERCENT_DONE )
+    -> set_schools_from_effect( 2 )
+    -> add_invalidate( CACHE_PLAYER_DAMAGE_MULTIPLIER )
+    -> set_stack_change_callback( [ this ]( buff_t*, int old_stacks, int new_stacks)
+          {
+            if ( new_stacks > old_stacks )
+              replenish_rune( as<unsigned int>( find_spell( 353823 ) -> effectN( 1 ).resource( RESOURCE_RUNE ) ), gains.final_sentence );
```

Comment:
Once shackle works again, you should check whether the rune gen works when you're already at full stacks of the damage buff and move the run gen to the dot spread if necessary.

Reply:
good catch. Thanks.